packages/pages/src/common/src/function/internal/types.ts

```diff
@@ -1,128 +1,88 @@
   name: string;
   /** The http event. */
-  event?: PluginEvent;
-  /** The function's name */
+  event: PluginEvent;
+  /** The default export's name */
   functionName: string;
 }
 
-export const convertServerlessFunctionModuleToServerlessFunctionModuleInternal =
-  (
-    serverlessFunctionFilepath: string,
-    serverlessFunctionModule: ServerlessFunctionModule<any>,
-    adjustForFingerprintedAsset: boolean
-  ): ServerlessFunctionModuleInternal<any> => {
-    const serverlessFunctionPath = parse(
-      serverlessFunctionFilepath,
-      adjustForFingerprintedAsset
-    );
-
-    let serverlessFunctionInternal;
+export const convertFunctionModuleToFunctionModuleInternal = (
+  functionFilepath: string,
+  functionModule: FunctionModule,
+  adjustForFingerprintedAsset: boolean
+): FunctionModuleInternal => {
+  const functionPath = parse(functionFilepath, adjustForFingerprintedAsset);
+  let functionInternal;
 
-    if (
-      serverlessFunctionFilepath.includes("/functions/http") &&
-      Object.keys(serverlessFunctionModule).length === 1
-    ) {
-      validateServerlessFunctionDefaultExport(
-        serverlessFunctionFilepath,
-        serverlessFunctionModule
-      );
-      const defaultPath = serverlessFunctionFilepath
-        .split("/functions/http/")[1]
-        .split(".")
-        .slice(-2)[0];
-      serverlessFunctionInternal = {
-        ...serverlessFunctionModule,
-        config: {
-          name: defaultPath,
-          functionName: serverlessFunctionPath.name,
-        },
-        path: serverlessFunctionFilepath,
-        filename: serverlessFunctionPath.base,
-        functionName: serverlessFunctionPath.name,
-        slug: defaultPath,
-        getPath: () => defaultPath,
-        default: serverlessFunctionModule.default,
-      };
-    } else {
-      validateServerlessFunctionModule(
-        serverlessFunctionFilepath,
-        serverlessFunctionModule
-      );
+  if (
+    functionFilepath.includes("/functions/http") &&
+    Object.keys(functionModule).length === 1
+  ) {
+    validateFunctionDefaultExport(functionFilepath, functionModule);
 
-      serverlessFunctionInternal = {
-        ...serverlessFunctionModule,
-        config:
-          convertServerlessFunctionConfigToServerlessFunctionConfigInternal(
-            serverlessFunctionPath.name,
-            serverlessFunctionModule.config
-          ),
-        path: serverlessFunctionFilepath,
-        filename: serverlessFunctionPath.base,
-        functionName: serverlessFunctionPath.name,
-        slug: serverlessFunctionModule.getPath
-          ? serverlessFunctionModule.getPath()
-          : "",
-      };
-    }
+    const defaultPath = functionFilepath
+      .split("/functions/http/")[1]
+      .split(".")
+      .slice(-2)[0];
 
-    validateServerlessFunctionModuleInternal(serverlessFunctionInternal);
+    functionInternal = {
+      ...functionModule,
+      config: {
+        name: defaultPath,
```

Comment:
The name should come from the FunctionConfig or the filename if http.

Reply:
similar to above -- we need to differentiate name in the case of `functions/http/get/index.ts` and `functions/http/post/index.ts` (unless we want to force all filenames to be unique)